src/core/Link/style.ts

```diff
@@ -3,12 +3,7 @@ import { Link, LinkProps as RawLinkProps } from "@mui/material";
 import { styled } from "@mui/material/styles";
 import { CommonThemeProps as StyleProps, getBorders } from "../styles";
 
-// (thuang): Support `component` prop
-// https://stackoverflow.com/a/66123108
-export type LinkProps<C extends React.ElementType = "a"> = RawLinkProps<
-  C,
-  { component?: C }
```

Comment:
do we want to remove this?  I'm curious if this would let us use a `button` instead when appropriate - the one I have has an onClick attached, so should probably be a `button` html element.

Reply:
Ooh you're right 😆 Just tested it locally and also needed to add a bit more type in Link component to make `component` prop work properly 👌 

And you're right that since there's onClick and it doesn't cause a page navigation, it should be a button element!

PTAL again thank you!